core/services/pg/advisory_lock.go

```diff
@@ -125,21 +116,16 @@ func (l *advisoryLock) logRetry(count int) {
 
 const checkAdvisoryLockStmt = `SELECT EXISTS (SELECT 1 FROM pg_locks WHERE locktype = 'advisory' AND pid = pg_backend_pid() AND (classid::bigint << 32) | objid::bigint = $1)`
 
-func (l *advisoryLock) loop() {
-	defer l.wg.Done()
-
+func (l *advisoryLock) loop(ctx context.Context) {
 	ticker := time.NewTicker(utils.WithJitter(l.checkInterval))
 	defer ticker.Stop()
 
-	ctxStop, cancel := utils.ContextFromChan(l.chStop)
-	defer cancel()
-
 	for {
 		select {
-		case <-l.chStop:
-			ctx, cancel := DefaultQueryCtx()
+		case <-ctx.Done():
```

Comment:
Shutdown is not synchronous any more, does it matter?

Reply:
Fixed